File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var checkIfExist = function(arr) {
    const st = new Set();
    for(const it of arr) {
        if(st.has(it*2) || st.has(it/2)) return true;
        st.add(it);
    }

    return false;
};

File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        HashSet<Integer> st = new HashSet<Integer>();
        for(int it : arr) {
            if(st.contains(it*2)) return true;
            if(it%2 == 0 && st.contains(it/2)) return true;
            st.add(it);
        }

        return false;
    }
}